torch_geometric/nn/pool/gm_pool.py

```diff
@@ -0,0 +1,291 @@
+        self.build_super_graph = build_super_graph
+        self.bipartite_k = bipartite_k
+        self.super_k = super_k
+        self.momentum = momentum
+        self.model = GaussianMixture(2)
+        if build_bipartite_graph:
+            self.bipartite_graph_construction = DynamicGraphConstruction(
+                bipartite_k, False, torch.exp)
+        if build_super_graph:
+            self.super_graph_construction = DynamicGraphConstruction(
+                super_k, True, torch.sigmoid)
+        self.register_buffer("score_cut", torch.tensor([0], dtype=torch.float),
+                             persistent=True)
+
+    def _get_quadratic_coeff(self, weight: float, mean: float,
+                             var: float) -> Tuple[float, float, float]:
+        """
+        find the coefficients of the quadratic equation that defines score cut
+        """
+        sigma = np.sqrt(var)
+        a = -0.5 / sigma**2
+        b = mean / sigma**2
+        c = -0.5 * mean**2 / sigma**2 - np.log(sigma) + np.log(weight)
+        return a, b, c
+
+    def _solve_quadratic_eq(self, a: float, b: float, c: float) -> float:
+        """
+        solve the quadratic equation
+        """
+        if b**2 > 4 * a * c:
+            return torch.as_tensor((-b + np.sqrt(b**2 - 4 * a * c)) / (2 * a),
+                                   dtype=torch.float)
+        else:
+            return torch.as_tensor(-b / (2 * a), dtype=torch.float)
+
+    def _determine_cut(self) -> float:
+        """
+        extract the parameters of Gaussians and solve for score cut. Always
+        pick the greater solution.
+        """
+        a1, b1, c1 = self._get_quadratic_coeff(
+            self.model.weights_[0].item(), self.model.means_[0].item(),
+            self.model.covariances_[0].item())
+        a2, b2, c2 = self._get_quadratic_coeff(
+            self.model.weights_[1].item(), self.model.means_[1].item(),
+            self.model.covariances_[1].item())
+        if self.model.means_[0][0] > self.model.means_[1][0]:
+            return self._solve_quadratic_eq(a1 - a2, b1 - b2, c1 - c2 - self.r)
+        else:
+            return self._solve_quadratic_eq(a2 - a1, b2 - b1, c2 - c1 - self.r)
+
+    def _get_clusters(self, labels: Tensor, batch: Tensor) -> Tensor:
+        """
+        filter out noises, i.e. components smaller than min_size.
+        """
+        _, inverse, counts = labels.unique(return_inverse=True,
+                                           return_counts=True)
+        noise_mask = counts[inverse] < self.min_size
+        is_noise = ~scatter(~noise_mask, batch, dim=0, reduce="any")
+        labels[is_noise[batch]] = batch[is_noise[batch]] + labels.max() + 1
+        noise_mask[is_noise[batch]] = False
+        labels[~noise_mask] = labels[~noise_mask].unique(
+            return_inverse=True)[1]
+        labels[noise_mask] = -1
+        return labels
+
+    def forward(
+        self, emb: Tensor, edges: Tensor, batch: Tensor
```

Comment:
(naming) `x` and `edge_index`, to be consistent with other pooling methods.

Reply:
Fixed